gpii/node_modules/flowManager/src/UserLogonStateChange.js

```diff
@@ -79,34 +82,53 @@ gpii.flowManager.getDeviceContext = function (deviceReporterDataSource, event, e
 };
 
 // A mixin grade for a matchMakingRequest request handler, supporting local user logon
-fluid.defaults("gpii.flowManager.userLogonStateChange.stateChangeHandler", {
+fluid.defaults("gpii.flowManager.userLogonHandling.stateChangeHandler", {
+    members: {
+        debounceTimeMs: 5000
+    },
     invokers: {
         getDeviceContext: { // TODO: remember to close GPII-1770 now this is correctly factored
             funcName: "gpii.flowManager.getDeviceContext",
             args: ["{flowManager}.deviceReporterDataSource", "{that}.events.onDeviceContext", "{request}.events.onError"]
         },
         startLifecycle: {
-            funcName: "gpii.flowManager.userLogonStateChange.startLifecycle",
-            args: [ "{flowManager}.lifecycleManager", "{arguments}.0", "{request}.events.onSuccess" ]
-                                                // final payload from matchmaking process
+            funcName: "gpii.flowManager.userLogonHandling.startLifecycle",
+            args: [ "{that}", "{flowManager}.lifecycleManager", "{arguments}.0", "{request}.events.onSuccess" ]
+                                                                // final payload from matchmaking process
         },
         logoutUser: {
-            funcName: "gpii.flowManager.userLogonStateChange.logoutUser",
-            args: ["{request}", "{flowManager}.lifecycleManager", "{arguments}.0"]
+            funcName: "gpii.flowManager.userLogonHandling.logoutUser",
+            args: ["{that}", "{request}", "{flowManager}.lifecycleManager", "{arguments}.0" ]
         },
         loginUser: {
-            funcName: "gpii.flowManager.userLogonStateChange.loginUser",
+            funcName: "gpii.flowManager.userLogonHandling.loginUser",
             args: ["{that}", "{arguments}.0"]
         },
         errorResponse: {
-            funcName: "gpii.flowManager.userLogonStateChange.errorResponse",
+            funcName: "gpii.flowManager.userLogonHandling.errorResponse",
             args: ["{request}", "{arguments}.0", "{arguments}.1"]
+        },
+        isBounce: {
+            funcName: "gpii.flowManager.userLogonHandling.isBounce",
+            args: ["{flowManager}.lifecycleManager", "{that}.debounceTimeMs", "{arguments}.0"] // userToken
         }
+    },
+    events: {
+        userLoginInitiated: null,
+        userLogoutInitiated: null,
+        userLoginComplete: null,
+        userLogoutComplete: null
+    },
+    listeners: {
+        "userLogoutInitiated.propagate": "{flowManager}.events.userLogoutInitiated.fire",
```

Comment:
Clearer just to inject these events down wholesale from the flowManager rather than relay them up like this